docs/excel/excel-add-ins-ranges-advanced.md

```diff
@@ -64,6 +64,96 @@ Your add-in will have to format the ranges to display the dates in a more human-
+
+    return context.sync();
+})
+```
+
+Sometimes the range doesn't have *any* cells with the targeted characteristic. If `getSpecialCells` doesn't find any, it throws an **ItemNotFound** error. This would divert the flow of control to a `catch` block/method, if there is one. If there isn't, the error halts the function. There may be scenarios in which throwing the error is exactly what you want to happen when there are no cells with the targeted characteristic.
+
+But in scenarios in which it is normal, but perhaps uncommon, for there to be no matching cells; your code should check for this possibility and handle it gracefully without throwing an error. For these scenarios, use the `getSpecialCellsOrNullObject` method and test the `RangeAreas.isNullObject` property. The following is an example. Note about this code:
+
+- The `getSpecialCellsOrNullObject` method always returns a proxy object, so it is never `null` in the ordinary JavaScript sense. But if no matching cells are found, the `isNullObject` property of the object is set to `true`.
+- It calls `context.sync` *before* it tests the `isNullObject` property. This is a requirement with all `*OrNullObject` methods and properties, because you always have to load and sync a property in order to read it. However, it is not necessary to *explicitly* load the `isNullObject` property. It is automatically loaded by the `context.sync` even if `load` is not called on the object. For more information, see [\*OrNullObject](https://docs.microsoft.com/office/dev/add-ins/excel/excel-add-ins-advanced-concepts#42ornullobject-methods).
+- You can test this code by first selecting a range that has no formula cells and running it. Then select a range that has at least one cell with a formula and run it again.
+
+```js
+Excel.run(function (context) {
+    var range = context.workbook.getSelectedRange();
+    var formulaRanges = range.getSpecialCellsOrNullObject("Formulas");
+    return context.sync()
+        .then(function() {
+            if (formulaRanges.isNullObject) {
+                console.log("No cells have formulas");
+            }
+            else {
+                formulaRanges.format.fill.color = "pink";
+            }
+        })
+        .then(context.sync);
+})
+```
+
+For simplicity, all other examples in this article use the `getSpecialCells` method instead of  `getSpecialCellsOrNullObject`.
+
+### Narrow the target cells with cell value types
+
+There is an optional second parameter, of enum type `Excel.SpecialCellValueType`, that further narrows down the cells to target. You can use it only when you pass either "Formulas" or "Constants" to `getSpecialCells` or `getSpecialCellsOrNullObject`. The parameter specifies that you only want cells with certain types of values. There are four basic types: "Error", "Logical" (which means boolean), "Numbers", and "Text". (The enum has other values besides these four which are discussed below.) The following is an example. About this code, note:
```

Comment:
- Suggest updating the first few sentences -> 
    The `Range.getSpecialCells()` and `Range.getSpecialCellsOrNullObject()` methods accept an optional second parameter that you can use to further narrow down the cells to target. This second parameter is an enum of type `Excel.SpecialCellValueType` and you can use it to specify that you only want cells that contain certain types of values.

- Suggest removing the sentence "This parameter is only valid when you pass either "Formulas" or "Constants" as the first parameter value -- and instead moving the [!NOTE] that contains this same info from the bottom of the section to immediately follow the first couple of sentences I've suggested above.

- Suggest adding an H4 heading `#### Test for a single cell value type` before you list the four basic types.

- Suggest presenting the four values as a list and rephrasing slightly. ->
    The `Excel.SpecialCellValueType` enum has these four basic types (in addition to the other combined values described later in this section):
        - "Error"
        - "Logical" (which means boolean)
        - "Numbers"
        - "Text"
        
- Suggest rephrasing "The following is an example." -> "The following example [describe what the example does]."